File: python/NeetCode150/5_top_k_ele.py
from collections import Counter
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        num_count = Counter(nums)
        output = []
        for i in range(0, k):
            biggest = (None, 0)
            for key in num_count:
                if num_count[key] > biggest[1]:
                    biggest = (key, num_count[key])
            num_count.pop(biggest[0])
            output.append(biggest[0])
        return output

File: c++/NeetCode150/5_top_k_ele.cpp
#include <unordered_map>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) 
    {
        unordered_map<int, int> num_count;
        for(auto& it: nums)
        {
            if (num_count.find(it) != num_count.end())
            {
                num_count[it]+=1;
            }
            else 
            {
                num_count[it] = 1;
            }
        }
        vector<int> output;
        for(int i=0; i<k; i++)
        {
            int biggest = 0;
            int index;
            for(auto& kv: num_count)
            {
                cout << kv.first << " " << kv.second << endl;
                if (kv.second > biggest)
                {
                    index = kv.first;
                    biggest = kv.second;
                }
            }
            output.push_back(index);
            num_count.erase(index);
        }
        return output;
    }
};
